### ISSUE TITLE
ListenBucketNotification Fails to Terminate Promptly

### ISSUE THREAD

**Issue by magaldima** [minio/minio-go#936]
**Problem**:
I am invoking the ListenBucketNotification function and passing in a `doneCh` that I created. Later, in my function, I am looking to stop listening to bucket notifications so I send a signal on that channel, however sending that signal blocks waiting on that signal to be read from the channel (on the Minio side). I think I believe I know what the issue is. In the block of code from the `ListenBucketNotification` function, we never recieve anything off the `doneCh` if there were not events received. I can verify that once I trigger an event, the doneCh message is received and the Listener stops. 

**Problem Code**: 

> api-notification.go

```
if len(notificationInfo.Records) > 0 {
	select {
	case notificationInfoCh <- notificationInfo:
	case <-doneCh:
		return
	}
}
```

**Expected**:
I would expect sending a signal on the `doneCh` to terminate the listening immediately, I don't think I should have to trigger another event just to make sure the doneCh message is received.

**Comment by harshavardhana** [comment]
> if len(notificationInfo.Records) > 0 {
> 	select {
> 	case notificationInfoCh <- notificationInfo:
> 	case <-doneCh:
> 		return
> 	}
> }

You are right this has to be handled outside.. @magaldima

**Comment by magaldima** [comment]
After the conditional, we can just have an additional select statement so it looks like this:
```
// Send notifications on channel only if there are events received.
if len(notificationInfo.Records) > 0 {
	select {
	case notificationInfoCh <- notificationInfo:
	case <-doneCh:
		return
	}
}
select {
case <-doneCh:
	return
}
```

**Comment by magaldima** [comment]
#937

**Comment by magaldima** [comment]
Should the conditional also check that the `notificationInfo.Err != nil`?

**Comment by harshavardhana** [comment]
Fixed by #940 - Thanks for the PR @magaldima
